File: app/src/main/java/com/specknet/pdiotapp/history/ActivityListAdapter.java
package com.specknet.pdiotapp.history;

import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.DiffUtil;
import androidx.recyclerview.widget.ListAdapter;


public class ActivityListAdapter extends ListAdapter<Activity, ActivityViewHolder> {

    public ActivityListAdapter(@NonNull DiffUtil.ItemCallback<Activity> diffCallback) {
        super(diffCallback);
    }

    @Override
    public ActivityViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return ActivityViewHolder.create(parent);
    }

    @Override
    public void onBindViewHolder(ActivityViewHolder holder, int position) {
        Activity current = getItem(position);
        holder.bind(current.getActivity());
    }

    static class WordDiff extends DiffUtil.ItemCallback<Activity> {

        @Override
        public boolean areItemsTheSame(@NonNull Activity oldItem, @NonNull Activity newItem) {
            return oldItem == newItem;
        }

        @Override
        public boolean areContentsTheSame(@NonNull Activity oldItem, @NonNull Activity newItem) {
            return oldItem.getActivity().equals(newItem.getActivity());
        }
    }
}

File: app/src/main/java/com/specknet/pdiotapp/history/ActivityListAdapter.kt
//package com.specknet.pdiotapp.history
//
//import android.view.ViewGroup
//import androidx.recyclerview.widget.DiffUtil
//import androidx.recyclerview.widget.ListAdapter
//
//
//internal class ActivityListAdapter(diffCallback: WordDiff) :
//    ListAdapter<Activity, ActivityViewHolder>(diffCallback) {
//    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ActivityViewHolder {
//        return ActivityViewHolder.create(parent)
//    }
//
//    override fun onBindViewHolder(holder: ActivityViewHolder, position: Int) {
//        val current: Activity? = getItem(position)
//        if (current != null) {
//            holder.bind(current.activityName)
//        }
//    }
//
//    internal class WordDiff : DiffUtil.ItemCallback<Activity>() {
//        override fun areItemsTheSame(oldItem: Activity, newItem: Activity): Boolean {
//            return oldItem === newItem
//        }
//
//        override fun areContentsTheSame(oldItem: Activity, newItem: Activity): Boolean {
//            return oldItem.activityName.equals(newItem.activityName)
//        }
//    }
//}